scripts/main.component.ts

```diff
@@ -20,18 +21,36 @@ import { REPLComponent, REPLDelegate } from "./repl.component"
   templateUrl: "../html/main.component.html",
   providers: [MenuService, PluginService]
 })
-export class MainComponent implements OnInit, MenuDelegate, REPLDelegate {
+export class MainComponent implements OnInit, MenuDelegate, REPLDelegate, EditorDelegate {
   constructor(private menu: MenuService, private pluginService: PluginService) {}
+  
+  selectedElements : Set<DrawableGraph | DrawableNode | DrawableEdge> =
+    new Set<DrawableGraph | DrawableNode | DrawableEdge>();
+  selectElement(element : DrawableGraph | DrawableNode | DrawableEdge) {
```

Comment:
My opinion is that the main component AKA document controller mostly just wires everything up together. I want to avoid doing serious work in it. The document controller WILL end up being massive, and I'd prefer not to clutter it any further. Otherwise we're going to end up with a god component that all other's depend on (indirectly in this case, but still.)